lib/bundler/runtime.rb

```diff
@@ -37,7 +37,17 @@ def setup(*groups)
 
         Bundler.rubygems.mark_loaded(spec)
         load_paths = spec.load_paths.reject {|path| $LOAD_PATH.include?(path) }
-        $LOAD_PATH.unshift(*load_paths)
+
+        # See Gem::Specification#add_self_to_load_path (since RubyGems 1.8)
+        insert_index = Gem.load_path_insert_index
```

Comment:
@Elffers this PR is totally acceptable without refactoring the method.

@drbrain I don't think Bundler should have an add_self_to_load_path either, at least not right now.

This PR only needs one change to be mergeable, and it is moving all references to `Gem` into the `rubygems_integration.rb` file. Something like this, I imagine:

```ruby
insert_index = Bundler.rubygems.load_path_insert_index

# meanwhile, in rubygems_integration.rb
def load_path_insert_index
  Gem.load_path_insert_index
end
```

Based on past experience, it's really important to keep references to RubyGems constants or methods isolated in the `rubygems_integration.rb` file so that we can change them in one central place when needed.